src/apis/emscripten/time.rs

```diff
@@ -64,37 +66,137 @@ pub extern "C" fn _clock_gettime(clk_id: c_int, tp: c_int, instance: &mut Instan
     0
 }
 
+#[repr(C)]
+struct guest_tm {
+    pub tm_sec: c_int, // 0
+    pub tm_min: c_int, // 4
+    pub tm_hour: c_int, // 8
+    pub tm_mday: c_int, // 12
+    pub tm_mon: c_int, // 16
+    pub tm_year: c_int, // 20
+    pub tm_wday: c_int, // 24
+    pub tm_yday: c_int, // 28
+    pub tm_isdst: c_int, // 32
+    pub tm_gmtoff: c_int, // 36
+    pub tm_zone: c_int, // 40
+}
+
+/// emscripten: _asctime
+pub extern "C" fn _asctime(time: u32, instance: &mut Instance) -> u32 {
+    debug!("emscripten::_asctime {}", time);
+
+    unsafe {
+        let date = &*(instance.memory_offset_addr(0, time as _) as *mut guest_tm);
+
+        let days = vec!["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
+        let months = vec!["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+
+        let day = if date.tm_mday < 10 {"  "} else {" "};
+        let hour = if date.tm_hour < 10 {" 0"} else {" "};
+        let min = if date.tm_min < 10 {":0"} else {":"};
+        let sec = if date.tm_sec < 10 {":0"} else {":"};
+        let year = 1900 + date.tm_year;
+
+        let time_str = format!(
+            // NOTE: The 14 accompanying chars are needed for some reason
+            "{} {}{}{}{}{}{}{}{}{} {}\0\0\0\0\0\0\0\0\0\0\0\0\0\0",
```

Comment:
Why so many `0`s?

Reply:
Exactly! I'm still researching why it segfaults if they are not there. 
I'm going to share my concerns on the dev channel.